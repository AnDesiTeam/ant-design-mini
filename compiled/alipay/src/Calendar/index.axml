<import-sjs
  from="./helper.sjs"
  name="helper" />

<import-sjs
  from="./scroll.sjs"
  name="scroll" />

<view
  class="ant-calendar {{ className ? className : '' }}"
  style="{{ style }}">
  <view class="ant-calendar-mark">
    <block
      a:for="{{ markItems }}"
      a:for-index="index"
      a:for-item="item">
      <view class="{{ helper.getMarkCellClassName(index, markItems) }}">
        <!-- display: inline -->
        <text>{{ item }}</text>
      </view>
    </block>
  </view>
  <block a:if="{{ !!elementSize }}">
    <view class="ant-calendar-sticky">
      <view class="ant-calendar-sticky-title">
        <slot name="calendarTitle">
          <view class="ant-calendar-title">
            {{ monthList[headerState].title }}
          </view>
        </slot>
      </view>
    </view>
  </block>
  <scroll-view
    scroll-y="{{ true }}"
    class="ant-calendar-body"
    data-elementsize="{{ elementSize }}"
    data-monthlist="{{ monthList }}"
    onScroll="{{ scroll.handleScroll }}"
    ref="handleRef">
    <block
      a:for="{{ monthList }}"
      a:for-index="index"
      a:for-item="currentMonth">
      <view class="ant-calendar-body-container">
        <view class="ant-calendar-title-container">
          <slot name="calendarTitle">
            <view class="ant-calendar-title">{{ currentMonth.title }}</view>
          </slot>
        </view>
        <view class="ant-calendar-cells">
          <block
            a:for="{{ currentMonth.cells }}"
            a:for-index="index"
            a:for-item="item">
            <block>
              <view
                class="{{ helper.getClassName(item, index) }}"
                data-time="{{ item }}"
                onTap="clickCell">
                <view class="ant-calendar-cell-container">
                  <view class="ant-calendar-cell-top">
                    <block a:if="{{ item.top }}">
                      <!-- display: inline -->
                      <text
                        class="ant-calendar-cell-top-text {{ item.top.className ? item.top.className : '' }}"
                        >{{ item.top.label }}</text
                      >
                    </block>
                  </view>
                  <view class="ant-calendar-cell-center">{{ item.date }}</view>
                  <view class="ant-calendar-cell-bottom">
                    <slot
                      name="cell-bottom"
                      cell="{{ item }}">
                      <block a:if="{{ item.bottom }}">
                        <!-- display: inline -->
                        <text class="{{ item.bottom.className }}"
                          >{{ item.bottom.label }}</text
                        >
                      </block>
                    </slot>
                  </view>
                </view>
              </view>
              <block a:if="{{ index % 7 !== 6 }}">
                <view
                  class="{{ helper.getSpaceClassName(index, currentMonth.cells) }}" />
              </block>
            </block>
          </block>
        </view>
      </view>
    </block>
  </scroll-view>
</view>
