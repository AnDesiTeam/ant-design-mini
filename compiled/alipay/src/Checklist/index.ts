import { ChecklistDefaultProps } from './props';
import mixinValue from '../mixins/value';
import fmtEvent from '../_util/fmtEvent';
import '../_util/assert-component2';

Component({
  props: ChecklistDefaultProps,
  mixins: [
    mixinValue({
      transformValue(val) {
        const value = val || [];
        return {
          needUpdate: true,
          value,
        };
      },
    }),
  ],
  methods: {
    onChange(item) {
      const { multiple, options, onChange } = this.props;
      const value = item.value;
      if (multiple) {
        let currentValue = this.getValue();
        if (currentValue.indexOf(value) > -1) {
          currentValue = currentValue.filter((v) => v !== value);
        } else {
          currentValue = [...currentValue, value];
        }
        if (!this.isControlled()) {
          this.update(currentValue);
        }
        if (onChange) {
          onChange(
            currentValue,
            options.filter((v) => currentValue.indexOf(v.value) > -1) as any,
            fmtEvent(this.props)
          );
        }
      } else {
        if (!this.isControlled()) {
          this.update(value);
        }
        if (onChange) {
          onChange(
            value,
            options.find((v) => v.value === value) as any,
            fmtEvent(this.props)
          );
        }
      }
    },
  },
});
