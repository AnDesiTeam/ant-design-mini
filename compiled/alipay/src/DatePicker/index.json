{
  "component": true,
  "lazyCodeLoading": "renderedComponents",
  "usingComponents": {
    "ant-picker": "../Picker/index"
  }
}
