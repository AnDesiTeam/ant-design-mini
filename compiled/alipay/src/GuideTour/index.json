{
  "component": true,
  "lazyCodeLoading": "renderedComponents",
  "usingComponents": {
    "mask": "../Mask/index",
    "ant-button": "../Button/index",
    "ant-icon": "../Icon/index"
  }
}
