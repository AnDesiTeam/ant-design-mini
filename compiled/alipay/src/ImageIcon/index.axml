<import-sjs
  from="./index.sjs"
  name="helper" />

<view class="ant-imageIcon {{ className ? className : '' }}">
  <block a:if="{{ helper.isIcon(image) }}">
    <icon
      type="{{ image }}"
      className="{{ className ? className + '-icon' : '' }}"
      style="{{ style }}" />
  </block>
  <block a:else>
    <image
      src="{{ image }}"
      class="{{ className ? className + '-image' : '' }}"
      style="{{ style || '' }}"
      mode="aspectFit" />
  </block>
</view>
