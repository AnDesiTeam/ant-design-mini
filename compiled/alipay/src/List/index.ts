import { mountComponent } from '../_util/component';
import { IListProps } from './props';

const List = () => {
  //
};

mountComponent<IListProps>(List, {
  radius: false,
  header: '',
  footer: '',
});
