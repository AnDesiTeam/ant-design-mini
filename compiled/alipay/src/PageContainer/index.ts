import equal from 'fast-deep-equal';
import { Component, getValueFromProps, triggerEventOnly } from '../_util/simply';
import { PageDefaultProps, BuiltinStatus } from './props';

Component(
  PageDefaultProps,
  {
    handleActionTap(e) {
      triggerEventOnly(this, 'actionTap', e);
    },
    handleSecondaryActionTap(e) {
      triggerEventOnly(this, 'secondaryActionTap', e);
    },
    updatePageStatus(prevProps: any, nextProps: any) {
      if (!equal(prevProps, nextProps)) {
        const { status, image, title, message } = nextProps;
        const updateData = {
          ...nextProps,
          // 自定义内容优先 status
          image: image || BuiltinStatus[status]?.image || '',
          title: title || BuiltinStatus[status]?.title || '',
          message: message || BuiltinStatus[status]?.message || ''
        };

        this.setData(updateData);
      }
    }
  },
  {},
  undefined,
  {
    didMount() {
      const props = getValueFromProps(this);
      this.updatePageStatus({}, props);
    },
    deriveDataFromProps(nextProps) {
      const props = getValueFromProps(this);
      this.updatePageStatus(props, nextProps);
    }
  }
);
