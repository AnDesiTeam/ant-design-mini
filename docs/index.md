---
title: Ant Design Mini
sidemenu: false
---

<code src="./components/home.tsx" inline="true"></code>

