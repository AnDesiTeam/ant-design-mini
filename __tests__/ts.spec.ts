import { getDocsRes,Resource } from "../scripts/docs/src";

describe("ts interface parse", () => {
    it('base', ()=>{
        const path =`${process.cwd()}/src/Tabs/TabItem`;
        const resource = new Resource();
        getDocsRes(path, resource);
        expect(JSON.stringify(resource)).toMatchSnapshot();
    })
});
