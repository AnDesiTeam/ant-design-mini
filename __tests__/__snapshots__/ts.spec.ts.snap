// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts interface parse base 1`] = `"{\\"class\\":[],\\"slot\\":[],\\"cssVar\\":[],\\"tips\\":[],\\"title\\":\\"\\",\\"desc\\":\\"标签栏内部元素，必须配合 Tabs 使用。\\",\\"api\\":{\\"prop\\":[{\\"description\\":\\"每一项 tab 内容\\",\\"default\\":\\"\\",\\"name\\":\\"tab\\",\\"types\\":\\"{ title: string; subTitle?: string; badge?: number; disabled?: boolean; }\\"},{\\"description\\":\\"类名\\",\\"default\\":\\"\\",\\"name\\":\\"className\\",\\"types\\":\\"string\\"}],\\"method\\":[]}}"`;
